pkg/resource/deploy/step.go

```diff
@@ -93,8 +93,7 @@ func (s *SameStep) Res() *resource.State { return s.new }
 func (s *SameStep) Logical() bool        { return true }
 
 func (s *SameStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error) {
```

Comment:
As discussed offline, there are some serious dragons around here w.r.t. to the exposure of new vs. old URNs via the `URN`, `Old`, `New`, `Type`, and `Provider` methods on `Step`. I'm not entirely sure what to _do_ about that yet, but ensuring that we consistently use the new URN to refer to a resource may be a good starting point.

Reply:
I updated to use `new` values for these wherever there is a valid `new, and thosse changes were pulled in independently in https://github.com/pulumi/pulumi/pull/2787.